frontend/src/metabase/common/components/EntityPicker/components/EntityPickerModal/TabsView.tsx

```diff
@@ -1,105 +1,38 @@
-import { useEffect, useMemo, useState } from "react";
-
 import { Icon, Tabs } from "metabase/ui";
-import type {
-  SearchRequest,
-  SearchResult,
-  SearchResultId,
-} from "metabase-types/api";
+import type { SearchResult, SearchResultId } from "metabase-types/api";
 
 import type { EntityTab, TypeWithModel } from "../../types";
 import {
   EntityPickerSearchResults,
   EntityPickerSearchTab,
 } from "../EntityPickerSearch";
 
-const computeInitialTab = <
-  Item extends TypeWithModel<SearchResultId, Model>,
-  Model extends string,
->({
-  initialValue,
-  tabs,
-  hasRecents,
-  defaultToRecentTab,
-}: {
-  initialValue?: Partial<Item>;
-  tabs: EntityTab<Model>[];
-  hasRecents: boolean;
-  defaultToRecentTab: boolean;
-}) => {
-  if (hasRecents && defaultToRecentTab) {
-    return { model: "recents" };
-  }
-  if (
-    initialValue?.model &&
-    tabs.some(tab => tab.model === initialValue.model)
-  ) {
-    return { model: initialValue.model };
-  } else {
-    return { model: tabs[0].model };
-  }
-};
-
 export const TabsView = <
   Id extends SearchResultId,
   Model extends string,
   Item extends TypeWithModel<Id, Model>,
 >({
-  tabs,
-  onItemSelect,
   searchQuery,
   searchResults,
   selectedItem,
-  initialValue,
-  defaultToRecentTab,
-  setShowActionButtons,
+  selectedTab,
+  tabs,
+  onItemSelect,
+  onTabChange,
 }: {
-  tabs: EntityTab<Model>[];
-  onItemSelect: (item: Item) => void;
   searchQuery: string;
   searchResults: SearchResult[] | null;
   selectedItem: Item | null;
-  initialValue?: Partial<Item>;
-  searchParams?: Partial<SearchRequest>;
-  defaultToRecentTab: boolean;
-  setShowActionButtons: (showActionButtons: boolean) => void;
+  selectedTab: Model | "recents" | "search";
+  tabs: EntityTab<Model | "recents" | "search">[];
+  onItemSelect: (item: Item) => void;
+  onTabChange: (model: Model | "recents" | "search") => void;
```

Comment:
I sorted these props alphabetically - thanks to this I noticed that `searchParams` was unused